libavcodec/libproresdecoder.c

```diff
@@ -0,0 +1,237 @@
+#include <proresdecoder.h>
+#include "libproresdecoder.h"
+#include "libavutil/pixfmt.h"
+#include "thread.h"
+
+int initAppleProResDecoder(struct AVCodecContext* avctx) {
+    ProResDecodeCodec* proresdecodecctx = avctx->priv_data;
+    proresdecodecctx->m_width = avctx->width; 
+    proresdecodecctx->m_height = avctx->height; 
+    // For now the downscale mode is universally 1x1
+    proresdecodecctx->downscale_mode = kPRDownscale_1x1;
+    
```

Comment:
For the reading of the tags and setting up accordingly, we follow the same approach as the FFmpeg prores decoder does: https://github.com/jaypadia-frame/FFmpeg/blob/master/libavcodec/proresdec2.c#L134